drivers/display/ssd1327.c

```diff
@@ -0,0 +1,396 @@
+		SSD1327_LINEAR_LUT,
+		SSD1327_SET_PRECHARGE_VOLTAGE,
+		SSD1327_DEFAULT_PRECHARGE_V,
+		SSD1327_SET_VCOMH,
+		SSD1327_VCOMH_VOLTAGE,
+		SSD1327_FUNCTION_SELECTION_B,
+		SSD1327_ENABLE_SECOND_PRECHARGE,
+		SSD1327_SET_COMMAND_LOCK,
+		SSD1327_UNLOCK_COMMAND,
+	};
+
+	return ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+}
+
+static inline int ssd1327_set_hardware_config(const struct device *dev)
+{
+	const struct ssd1327_config *config = dev->config;
+
+	uint8_t cmd_buf[] = {
+		SSD1327_SET_DISPLAY_START_LINE,
+		config->start_line,
+		SSD1327_SET_DISPLAY_OFFSET,
+		config->display_offset,
+		SSD1327_SET_NORMAL_DISPLAY,
+		SSD1327_SET_MULTIPLEX_RATIO,
+		config->multiplex_ratio,
+		SSD1327_SET_FUNCTION_A,
+		SSD1327_ENABLE_VDD,
+	};
+
+	return ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+}
+
+static int ssd1327_resume(const struct device *dev)
+{
+	uint8_t cmd_buf[] = {
+		SSD1327_DISPLAY_ON,
+	};
+
+	return ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+}
+
+static int ssd1327_suspend(const struct device *dev)
+{
+	uint8_t cmd_buf[] = {
+		SSD1327_DISPLAY_OFF,
+	};
+
+	return ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+}
+
+static int ssd1327_set_display(const struct device *dev)
+{
+	const struct ssd1327_config *config = dev->config;
+
+	uint8_t cmd_buf[] = {
+		SSD1327_SET_COLUMN_ADDR,
+		0,
+		config->width - 1,
+		SSD1327_SET_ROW_ADDR,
+		0,
+		config->height - 1,
+		SSD1327_SET_SEGMENT_MAP_REMAPED,
+		config->remap_value,
+	};
+
+	return ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true);
+}
+
+static int ssd1327_write_default(const struct device *dev, const uint16_t x, const uint16_t y,
+				 const struct display_buffer_descriptor *desc, const void *buf,
+				 const size_t buf_len)
+{
+	uint8_t cmd_buf[] = {
+		SSD1327_SET_COLUMN_ADDR,
+		x,
+		x + desc->width - 1,
+		SSD1327_SET_ROW_ADDR,
+		y,
+		y + desc->height - 1
+	};
+
+	if (ssd1327_write_bus(dev, cmd_buf, sizeof(cmd_buf), true)) {
+		LOG_ERR("Failed to write command");
+		return -1;
+	}
+
+	return ssd1327_write_bus(dev, (uint8_t *)buf, buf_len, false);
+}
+
+static int ssd1327_write(const struct device *dev, const uint16_t x, const uint16_t y,
+			 const struct display_buffer_descriptor *desc, const void *buf)
+{
+	size_t buf_len;
+
+	if (desc->pitch < desc->width) {
+		LOG_ERR("Pitch is smaller then width");
+		return -1;
+	}
+	/* Following the datasheet, in the GDDRAM, two segment are split in one register */
+	buf_len = MIN(desc->buf_size, desc->height * desc->width / 2);
+	if (buf == NULL || buf_len == 0U) {
+		LOG_ERR("Display buffer is not available");
+		return -1;
+	}
+
+	if (desc->pitch > desc->width) {
+		LOG_ERR("Unsupported mode");
+		return -1;
+	}
+
+	if ((y & 0x7) != 0U) {
+		LOG_ERR("Unsupported origin");
+		return -1;
+	}
+
+	LOG_DBG("x %u, y %u, pitch %u, width %u, height %u, buf_len %u", x, y, desc->pitch,
+		desc->width, desc->height, buf_len);
+
+	return ssd1327_write_default(dev, x, y, desc, buf, buf_len);
```

Comment:
Slight nit- but why use the `write_default` function here? The SSD1306 driver (which this driver seems based on) seems to have this because it can support a SH1106 display controller as well